src/nvim/eval/vars.c

```diff
@@ -1897,6 +1897,45 @@ static OptVal tv_to_optval(typval_T *tv, const char *option, uint32_t flags, boo
   return value;
 }
 
+/// Consume an option value and convert it to typval.
```

Comment:
"consume" implies that some state is being advanced (e.g. in a stream/queue/etc), is that the case here?

Reply:
Well, in this case, it just means it doesn't copy the value and instead uses the memory allocated by the OptVal.

Reply:
Sounds more like "read"? Should avoid words that have extra meaning unless the extra meaning is relevant.